### PR TITLE
Add Skew T plotting capabilities

### PR DESCRIPTION
**Pull request by kevindougherty-noaa**
There is a need to plot sounding data on a skew-T log-p plot. This PR adds that capability by introducing the proper projection and plotting the data correctly.

### CODE REVIEW

**Comment by kevindougherty-noaa** [comment]
@CoryMartin-NOAA Would you mind having a look at why the tests are failing? I don't quiet understand why it's doing that.

File: src/emcpy/plots/create_plots.py [hunk 1]
```diff
@@ -11,7 +11,10 @@
 from scipy.interpolate import interpn
 from cartopy.mpl.ticker import LongitudeFormatter, LatitudeFormatter
 from matplotlib.offsetbox import OffsetImage, AnchoredOffsetbox
+from matplotlib.ticker import MultipleLocator, NullFormatter, ScalarFormatter
+from matplotlib.projections import register_projection
 from emcpy.plots.map_tools import Domain, MapProjection
+from emcpy.plots.skewt_projection import SkewXAxes
```
Comment (CoryMartin-NOAA) [inline src/emcpy/plots/create_plots.py:17]: @kevindougherty-noaa where does this come from? There is no `skewt_projection` file in this PR. Should there be?

Comment (kevindougherty-noaa) [reply src/emcpy/plots/create_plots.py:17]: That would make sense lol. Oversight, just added.

File: examples/SkewT.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import sys
+sys.path.append('/scratch1/NCEPDEV/da/Kevin.Dougherty/emcpy/src/')
```
Comment (CoryMartin-NOAA) [inline examples/SkewT.py]: remove this?

File: src/emcpy/plots/map_plots.py [hunk 3]
```diff
@@ -81,3 +85,30 @@ def __init__(self, latitude, longitude, data):
         self.vmax = None
         self.alpha = None
         self.colorbar = False
+
+
+class MapContourf:
```
Comment (CoryMartin-NOAA) [inline src/emcpy/plots/map_plots.py]: why are these changed?

File: src/emcpy/plots/skewt_projection.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
+from contextlib import ExitStack
+
+from matplotlib.axes import Axes
+import matplotlib.transforms as transforms
+import matplotlib.axis as maxis
+import matplotlib.spines as mspines
+
+
+# The sole purpose of this class is to look at the upper, lower, or total
```
Comment (CoryMartin-NOAA) [inline src/emcpy/plots/skewt_projection.py:13]: was most of this "borrowed" from the matplotlib examples? if so, perhaps add a reference somewhere?

Comment (kevindougherty-noaa) [reply src/emcpy/plots/skewt_projection.py:13]: All of it was borrowed from here: https://matplotlib.org/stable/gallery/specialty_plots/skewt.html. MetPy did the same, but I will definitely add a reference.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by nicholasesposito — APPROVED** [review]
